### PR TITLE
Prefill check

### LINKED ISSUE #27: Apply automatic xml file and media check on prefill method as well
Currently, prefillAndOpenForm does not check for XML and media file availability before opening a form. Enable this feature in the corresponding method.

### PR DESCRIPTION
**Pull request by prabs3257**
Closes #27 

## Description

Please provide a brief description of the changes in this PR.

Added xml file and media check on prefill method

### Changes

List the changes done to fix a bug or introduce a new feature.

Added a new function in ODKFormHandler.kt

## Checklist

Please check the boxes below that apply:

- [X] All tests pass locally OR confirm CircleCI build passes.
- [X] Documentation has been updated or created, if applicable.

## How to Test
Please include any additional information or notes about testing here, if applicable.

## For UI-specific PRs only
If your PR includes UI-related changes, then:

Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes.   
Add a video showing the full UX flow.

### CODE REVIEW

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt [hunk 1]
```diff
@@ -72,6 +72,35 @@ class ODKFormsHandler @Inject constructor(
         openForm(form, context)
     }
 
+
+    private fun checkForm(formId: String, tagValueMap: HashMap<String, String>){
```
Comment (chinmoy12c) [inline odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt]: Hey @prabs3257 I see there is some code repetition in this function [here](https://github.com/Samagra-Development/odk-collect-extension/blob/3a0017b9ecfd3d8beafc2812d2f7f7eb1e24a52d/odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt#L201). Can you extract out the form files checking code into a single function and call that function both from prefillForm() and openForm().

**Review by chinmoy12c — CHANGES_REQUESTED** [review]

**Comment by prabs3257** [comment]
Heyy @chinmoy12c I have created a single check function now and used It in the prefillAndOpenForm() and openForm(). Please let me know if there are any additional changes needed 😃

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt [hunk 2]
```diff
@@ -275,7 +287,16 @@ class ODKFormsHandler @Inject constructor(
                     else -> {}
                 }
             })
-            prefillForm(formId, tagValueMap)
+            val form = formsDatabaseInteractor.getLatestFormById(formId)
+            if(checkForm(form)){
+                prefillForm(formId,tagValueMap);
+            }else{
+                formsNetworkInteractor.downloadFormById(formId, object : FileDownloadListener {
```
Comment (chinmoy12c) [inline odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt:294]: What happens if the download fails, how is that being handled?

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt [hunk 3]
```diff
@@ -208,21 +227,14 @@ class ODKFormsHandler @Inject constructor(
...
-                }
-                else {
-                    requiredForm.formMediaPath?.let { File(it).deleteRecursively() }
-                    xmlFile.delete()
-                    formsDatabaseInteractor.deleteByFormId(formId)
-                    downloadAndOpenForm(formId, context)
-                }
-            }
+
+
```
Comment (chinmoy12c) [inline odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt:237]: nit remove extra newlines.

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt [hunk 4]
```diff
@@ -208,21 +227,14 @@ class ODKFormsHandler @Inject constructor(
                 }
             }
             val requiredForm = formsDatabaseInteractor.getLatestFormById(formId)
-            if (requiredForm == null) {
+
+            if(checkForm(requiredForm)){
+                openFormWithFormId(formId, context)
+            }else{
```
Comment (chinmoy12c) [inline odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt:233]: nit `} else {`

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt [hunk 5]
```diff
@@ -72,6 +72,25 @@ class ODKFormsHandler @Inject constructor(
...
+
+    private fun checkForm(form: Form?) : Boolean{
+
+        if(form == null){
+            return false;
+        }else{
+            val xmlFile = File(form.formFilePath)
+            if (xmlFile.exists() && (form.formMediaPath == null || mediaExists(form))) {
+                return true;
+            }else {
```
Comment (chinmoy12c) [inline odk/extension/src/main/java/io/samagra/odk/collect/extension/handlers/ODKFormsHandler.kt:84]: nit `} else {`

**Review by chinmoy12c — CHANGES_REQUESTED** [review]

**Comment by chinmoy12c** [comment]
Closed in #64

**Pull request closed without merging**